frontend/src/utils/download.js

```diff
@@ -0,0 +1,31 @@
+import * as moment from 'moment';
+import html2canvas from 'html2canvas';
+
+export const generateDownloadFileName = (exportName, ext) => (
+  `${exportName}_${moment().format('YYYYMMDDHHmmss')}.${ext}`
+);
+
+export const downloadFile = (fileName, dataURL) => {
+  const downloadAnchorNode = document.createElement('a');
+  downloadAnchorNode.setAttribute('href', dataURL);
+  downloadAnchorNode.setAttribute('download', fileName);
+  document.body.appendChild(downloadAnchorNode); // required for firefox
+  downloadAnchorNode.click();
+  downloadAnchorNode.remove();
+};
+
+export const downloadObjectAsJson = (exportObj, exportName) => {
+  const fileName = generateDownloadFileName(exportName, 'json');
+  const dataURL = `data:text/json;charset=utf-8,${encodeURIComponent(JSON.stringify(exportObj))}`;
+  downloadFile(fileName, dataURL);
+};
+
+export const downloadChartAsPng = async (chartDOMNode, exportName) => {
```

Comment:
Is aysnc & await available? I don't know that `@babel/env` supports it. There is no problem if possible.

Reply:
Yes. I confirmed babel transpiles async/await correctly with current config.